filebeat/docs/modules/cisco.asciidoc

```diff
@@ -444,6 +445,86 @@ Maximum duration before AWS API request will be interrupted. Default to be 120 s
 
 :fileset_ex!:
 
+[float]
+==== `amp` fileset settings
+
+The Cisco AMP fileset focuses on collecting events from your Cisco AMP/Cisco Secure Endpoint API.
+
+To configure the Cisco AMP fileset you will need to retrieve your `client_id` and `api_key` from the AMP dashboard.
+For more information on how to retrieve these credentials, please reference the https://api-docs.amp.cisco.com/api_resources?api_host=api.amp.cisco.com&api_version=v1[Cisco AMP API documentation].
+
+The URL configured for the API depends on which region your AMP is located, currently there is 3 choices:
+- api.amp.cisco.com
+- api.apjc.amp.cisco.com
+- api.eu.amp.cisco.com
+
+If new endpoints are added by Cisco in the future, please reference the API URL list located at the https://api-docs.amp.cisco.com/[Cisco AMP API Docs].
+
+Example config:
+
+[source,yaml]
+----
+- module: cisco
+  amp:
+    enabled: true
+    var.input: httpjson
+    var.url: https://api.amp.cisco.com/v1/events
+    var.client_id: 123456
+    var.api_key: sfda987gdf90s0df0
+----
+
+When starting up the Filebeat module for the first time, you are able to configure how far back you want Filebeat to collect existing events from.
+It is also possible to select how often Filebeat will check the Cisco AMP API. Another example below which looks back 200 hours and have a custom timeout:
+
+[source,yaml]
+----
+- module: cisco
+  amp:
+    enabled: true
+    var.input: httpjson
+    var.url: https://api.amp.cisco.com/v1/events
+    var.client_id: 123456
+    var.api_key: sfda987gdf90s0df0
+    var.first_interval: 200h
+    var.interval: 60m
+    var.request_timeout: 120s
+    var.limit: 100
+
+----
+
+*`var.input`*::
+
+The input from which messages are read. Supports httpjson.
+
+*`var.url`*::
+
+The URL to the Cisco AMP API endpoint, this url value depends on your region. It will be the same region as your Cisco AMP Dashboard URL.
+
+*`var.client_id`*::
+
+The ID for the user account used to access the API.
+
+*`var.api_key`*::
+
+The API secret used together with the related client_id.
+
+*`var.request_timeout`*::
+
+When handling large influxes of events, especially for large enterprises, the API might take longer to respond. This value is to set a custom
+timeout value for each request sent by Filebeat.
+
+*`var.first_interval`*::
+
+How far back you would want to collect events the first time the Filebeat module starts up. Supports amount in hours.
```

Comment:
Is this really only supported in hours? Normally we accept values like this using `time.Duration` so the value can be given in any format.

Reply:
That is a good point. Initially I set this mostly because we cannot define anything larger than hours, and usually the interval would not need to be less than hour(s). But I can add that it supports minutes and seconds as well.